src/pyfmi/fmi_util.pyx

```diff
@@ -151,20 +154,19 @@ cpdef prepare_data_info(np.ndarray[int, ndim=2] data_info, list sorted_vars, lis
             data_info[0,i] = last_data_matrix
             data_info[1,i] = alias*last_index
         else:
-            variability = var.variability
             last_vref   = var.value_reference
-            data_type   = var.type
 
-            if variability == _FMI_PARAMETER or variability == _FMI_CONSTANT:
+            if var.variability == _FMI_PARAMETER or var.variability == _FMI_CONSTANT:
                 last_data_matrix = 1
                 index_fixed = index_fixed + 1
                 last_index = index_fixed
 
-                if data_type == _FMI_REAL:
+                # Make make sure we add the parameters to the correct list of value references
+                if var.type == FMI_REAL:
                     param_real.append(last_vref)
-                elif data_type == _FMI_INTEGER or data_type == _FMI_ENUMERATION:
+                elif (var.type == FMI_INTEGER) or (var.type == FMI_ENUMERATION):
```

Comment:
I see you changed from the `_FMI_TYPES` to `FMI_TYPE` here; If this doesn't cause any issues, we should maybe fully apply in this function and remove the definitions up top.

Reply:
Yes, fully getting rid of the _ prefix; not sure I see the point in them.